src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt

```diff
@@ -36,239 +51,475 @@ import javax.swing.tree.DefaultTreeModel
+                            descriptionPanel.add(scrollPane, BorderLayout.CENTER)
+                            //descriptionPanel.add(DeepCodeConfigForm().root, BorderLayout.CENTER)
+
+                            val textRange = suggestion.ranges.firstOrNull()
+                                ?: throw IllegalArgumentException(suggestion.ranges.toString())
+                            if (psiFile.virtualFile.isValid) {
+                                // jump to Source
+                                PsiNavigationSupport.getInstance().createNavigatable(
+                                    project,
+                                    psiFile.virtualFile,
+                                    textRange.start
+                                ).navigate(false)
+
+                                // highlight(by selection) suggestion range in source file
+                                val editor = FileEditorManager.getInstance(project).selectedTextEditor
+                                editor?.selectionModel?.setSelection(textRange.start, textRange.end)
+                            }
+                        }
+                        else -> {
+                            displayEmptyDescription()
+                        }
+                    }
+                }
+                descriptionPanel.revalidate()
+                descriptionPanel.repaint()
+            }
+        }
 
         project.messageBus.connect(this)
-            .subscribe(SnykCliScanListener.CLI_SCAN_TOPIC, object: SnykCliScanListener {
+            .subscribe(SnykScanListener.SNYK_SCAN_TOPIC, object : SnykScanListener {
 
-            override fun scanningStarted() =
-                ApplicationManager.getApplication().invokeLater { displayScanningMessage() }
+                override fun scanningStarted() =
+                    ApplicationManager.getApplication().invokeLater { displayScanningMessageAndUpdateTree() }
 
-            override fun scanningFinished(cliResult: CliResult) =
-                ApplicationManager.getApplication().invokeLater { displayVulnerabilities(cliResult) }
+                override fun scanningCliFinished(cliResult: CliResult) {
+                    currentCliResults = cliResult
+                    ApplicationManager.getApplication().invokeLater { displayVulnerabilities(cliResult) }
+                }
 
-            override fun scanError(cliError: CliError) =
-                ApplicationManager.getApplication().invokeLater { displayError(cliError) }
-        })
+                override fun scanningSnykCodeFinished(snykCodeResults: SnykCodeResults) =
+                    ApplicationManager.getApplication().invokeLater { displaySnykCodeResults(snykCodeResults) }
+
+                override fun scanError(cliError: CliError) =
+                    ApplicationManager.getApplication().invokeLater {
+                        if (cliError.message == "Authentication failed. Please check the API token on https://snyk.io") {
```

Comment:
This will work for producation SaaS only (and not for dev or on-prem instances).
How about don't match `https://snyk.io`?

Reply:
check CLI output with custom endpoint

Reply:
fixed. Previously it seems there always have be the link to `https://snyk.io` in that error message even with custom endpoint. It's not like that now, probably I did something wrong.